docs/source/onedal/algorithms/ensembles/decision-forest.rst

```diff
@@ -0,0 +1,218 @@
+.. |i_dense| replace:: `Dense <df_i_math_dense_>`_
+.. |i_hist| replace:: `Hist <df_i_math_hist_>`_
+.. |i_input| replace:: `infer_input <df_i_api_input_>`_
+.. |i_result| replace:: `infer_result <df_i_api_result_>`_
+.. |i_op| replace:: `infer(...) <df_i_api_>`_
+
+=============== ============= ============= ======== =========== ============
+ **Operation**  **Computational methods**     **Programming Interface**
+--------------- --------------------------- ---------------------------------
+   |t_math|        |t_dense|     |t_hist|    |t_op|   |t_input|   |t_result|
+   |i_math|        |i_dense|     |i_hist|    |i_op|   |i_input|   |i_result|
+=============== ============= ============= ======== =========== ============
+
+------------------------
+Mathematical formulation
+------------------------
+
+.. _df_t_math:
+
+Training
+--------
+Given :math:`n` feature vectors :math:`X=\{x_1=(x_{11},\ldots,x_{1p}),\ldots,x_n=(x_{n1},\ldots,x_{np})\}` of
+size :math:`p`, their non-negative observation weights :math:`W=\{w_1,\ldots,w_n\}` and :math:`n` responses :math:`Y=\{y_1,\ldots,y_n\}` (for classification, :math:`y_i \in \{0, \ldots, C-1\}`, where :math:`C` is the number of classes), the problem is to build a decision forest classification or regression model.
+
+During the training stage, :math:`M` independent classification or regression trees are created using the following:
+
+#. New training set generated from the original one by sampling uniformly and with replacement
+   (bootstrapping).
+#. Impurity metric :math:`I` and impurity reduction
+   :math:`I_{\mathrm{reduction}}` for splitting tree's nodes, calculated as follows:
+    - Gini impurity for classification:
+        - without observation weights: :math:`I(D)=1-\sum_{i=1}^{C}{p_i^2},` where :math:`p_i` is the fraction of observations in subset :math:`D` that belong to the :math:`i`-th class.
+        - with observation weights: :math:`I(D)=1-\sum_{i=1}^{C}{p_i^2},` where :math:`p_i` is the weighted fraction of observations in subset :math:`D` that belong to the :math:`i`-th class,
+	  :math:`p_i=(\sum_{d \in \{d \in D | y_{d}=i\}}w_d)/\sum_{d \in D}w_d,` and :math:`w_d` is a weight of observation :math:`d`. 
+    - Mean-Square Error (MSE) for regression: 
+	- without observation weights: :math:`I(D)=\frac{1}{N} \sum_{i=1}^{N}{(y_i - \bar{y})^2},` where :math:`N=|D|` and :math:`\bar{y}=\frac{1}{N} \sum_{i=1}^{N}y_i`.
+	- with observation weights: :math:`I(D)=\frac{1}{W(D)} \sum_{i=1}^{N}w_i{(y_i - \bar{y})^2},` where :math:`N=|D|`, :math:`\bar{y}=\sum_{i=1}^{N}w_{i}y_{i},`, :math:`W(D)=\sum_{i=1}^{N}w_{i},` and :math:`w_i` is a weight of observation :math:`i`.
+    - :math:`I_{\mathrm{reduction}}={I} - (N_{\mathrm{left}}/N_{\mathrm{parent}}*I_{left} + N_{\mathrm{right}}/N_{\mathrm{parent}}*I_{\mathrm{right}}),` where :math:`N_{\mathrm{left}}` and :math:`N_{\mathrm{right}}` are the number of observations in the node on the corresponding side of the split.
```

Comment:
tried to addressed @rlnx's comments from above, please check indentation:

```suggestion
   - Gini impurity for classification:
     - without observation weights: :math:`I(D)=1-\sum_{i=1}^{C}{p_i^2},` where :math:`p_i` is the fraction of observations in subset :math:`D` that belong to the :math:`i`-th class.
     - with observation weights: :math:`I(D)=1-\sum_{i=1}^{C}{p_i^2},` where :math:`p_i` is the weighted fraction of observations in subset :math:`D` that belong to the :math:`i`-th class, computed as follows:
	   
       .. math:
          p_i=(\sum_{d \in \{d \in D | y_{d}=i\}}w_d)/\sum_{d \in D}w_d
          
       where :math:`w_d` is a weight of observation :math:`d`.
   - Mean-Square Error (MSE) for regression: 
	 - without observation weights: :math:`I(D)=\frac{1}{N} \sum_{i=1}^{N}{(y_i - \bar{y})^2},` where :math:`N=|D|` and :math:`\bar{y}=\frac{1}{N} \sum_{i=1}^{N}y_i`.
	 - with observation weights: :math:`I(D)=\frac{1}{W(D)} \sum_{i=1}^{N}w_i{(y_i - \bar{y})^2},` where :math:`N=|D|`, :math:`\bar{y}=\sum_{i=1}^{N}w_{i}y_{i},`, :math:`W(D)=\sum_{i=1}^{N}w_{i},` and :math:`w_i` is a weight of observation :math:`i`.
     - :math:`I_{\mathrm{reduction}}` is computed as follows:
       
       .. math:
          I_{\mathrm{reduction}}={I} - (\frac{N_{\mathrm{left}}}{N_{\mathrm{parent}}} I_{left} + \frac{N_{\mathrm{right}}}{N_{\mathrm{parent}}} I_{\mathrm{right}}) 
       
       where :math:`N_{\mathrm{left}}` and :math:`N_{\mathrm{right}}` are the number of observations in the node on the corresponding side of the split.
```

Reply:
added, thank you